mysite/scrapers/scraperDomRiaScraper.py

```diff
@@ -1,22 +1,107 @@
-from scraperParentClass import WebScraper
-from bs4 import BeautifulSoup
-import requests
+import re
+import datetime
 from typing import Dict, Optional
+from bs4 import BeautifulSoup
+
+from mysite.scrapers.scraperParentClass import WebScraper
+from mysite.service.databasehandler import DatabaseHandler
 
 class DomRiaScraper(WebScraper):
 
-    def extract_data(self, soup: BeautifulSoup) -> Dict[str, Optional[str]]:
+    def __init__(self, listing_url: str):
+        super().__init__(website_url=listing_url)
+        self.listing_url = listing_url
+
+    
+    def scrape_property_details(self) -> Dict:
+
+        response = self.get_page()
+        if not response:
+            return {}
+        
+
+        soup = BeautifulSoup(response.text, "html.parser")
+        
+        property_details = {
+            'url': self.listing_url,
+            'source_website': 'dom.ria.com',
+            'created_at': datetime.datetime.now(),
+            'last_checked_at': datetime.datetime.now()
+        }
+
         deleted_tag = soup.find("span", class_="size24 bold")
         if deleted_tag and "видалено" in deleted_tag.text.lower():
-            availability = "deleted"
-            normalized_price = None
+            property_details['availability'] = 'deleted'
+
         else:
-            availability = "available"
+            property_details['availability'] = 'available'
             price_tag = soup.find("b", class_="size30")
             raw_price = price_tag.text.strip() if price_tag else None
             normalized_price = self.normalize_price(raw_price)
-        return {
-            "url": self.website_url,
-            "price": normalized_price,
-            "availability": availability
-        }
+
+            if normalized_price is None:
+                cleaned_price = "0"
+            else:
+                cleaned_price = re.sub(r'\D', '', normalized_price)
```

Comment:
Could you please elaborate what's the point of using `sub` here?
I am not a python export, but from the [documentation ](https://docs.python.org/uk/3.13/library/re.html#re.sub) (more newcomer freindly doc and example - [link](https://www.geeksforgeeks.org/re-sub-python-regex/). Overall it looks like it is mostly used for the substitution of the substrings. But I am not sure why we need substitution and not a simple search by regular expression?
Or is it to actually to clean up price from all the garbage like spaces, currency and etc?